server/src/controllers/hostingRequests.ts

```diff
@@ -0,0 +1,355 @@
+})
+
+// PZD-10
+// zatwierdz badz odrzuc prowadzacego prowadzacego
+// localhost:8000/hrequests/resolve/blabla
+hostingRequestRouter.put('/resolve/:id', async (request: Request, response: Response) => {
+  const id = parseInt(request.params.id)
+  const status = request.body.object.status
+
+  if(!validateValues(status, Status, response)) return
+
+  const connection = await getConnection();
+  const hRequestRepository = await connection.getRepository(HostingRequest)
+
+  await getConnection()
+    .createQueryBuilder()
+    .update(HostingRequest)
+    .set({status})
+    .where("id = :id", { id })
+    .execute()
+    .then(async () => {
+      if(status === "accepted") {
+        const hr = await hRequestRepository.findOne({id})
+
+        if(hr === undefined) {
+          return response.status(500).json({
+            status: "failure",
+            message: "hosting request not found"
+          })
+        }
+
+        const host = hr.user
+        await getConnection()
+        .createQueryBuilder()
+        .update(Class)
+        .set({host})
+        .where("groupKey = :groupKey", { groupKey: hr.class.groupKey })
+        .execute()
+        .then(() => {
+          return response.status(200).json({
+            status: "success"
+          })
+        })
+        .catch(error => {
+          logger.error(error)
+          return response.status(500).json({
+            status: "failure",
+            message: error.message
+          })
+        });
+      }
+    })
+    .catch(error => {
+      logger.error(error)
+      return response.status(500).json({
+        status: "failure",
+        message: error.message
+      })
+    });
+})
+
+// PZD-10
+// zapiszMnieNaKursy(objects: Classes[]): void
+// localhost:8000/hrequests
+hostingRequestRouter.post('/plan', async (request: Request, response: Response) => {
+  const objects = request.body.objects
+  const username = request.body.username
+
+  const connection = await getConnection();
+  const classRepository = connection.getRepository(Class)
+  const userRepository = connection.getRepository(User)
+
+  let newhostingRequests = []
+  for (let groupKey of objects.map((c: { groupKey: string; }) => c.groupKey)) {
+    const clas = await classRepository.findOne({groupKey});
+    const user = await userRepository.findOne({username});
+
+    if (clas != undefined && user != undefined) {
+      newhostingRequests.push({
+        status: "pending",
+        user,
+        class: clas
+      })
+    }
+  }
+
+  insertObjectIntoTable(newhostingRequests, HostingRequest, response);
+})
+
+//PZD27
+//rejectHostingRequests
+hostingRequestRouter.put('/reject', async (request: Request, response: Response) => {
+  const objects = request.body.objects
+  let ids = objects.map((o: { id: any; }) => o.id)
+
+  const connection = await getConnection();
+  const hrRepository = connection.getRepository(HostingRequest)
+
+  ids.array.forEach(async (id: any) => {
```

Comment:
i see some data consistency issue possibility, let me know if you have time so we can speak about it.